getline.ts/src/blockchain.ts

```diff
@@ -126,47 +147,73 @@ export class GetlineBlockchain {
     }
 
     /**
-     * Waits until a given Web3 contract is deployed on the blockchain.
+     * Waits until a given transaction hash is present on the blockchain.
      *
-     * This is a hack that had to be written for when web3.ether.contract.new 
+     * This is a hack that had to be written for when web3.ether.contract.new
      * fails to run the creation callback for the second time. When we upgrade
      * to web3 1.0 or another library we should be able to get rid of this.
      *
-     * @param T Web3 contract type.
-     * @param contract Web3 contract object to wait on.
-     * @returns Address of new contract.
+     * @param hash Hash of transaction to look for.
      */
-    private async waitTxReceipt(contract: Web3.ContractInstance): Promise<Address> {
-        return new Promise<Address>((resolve, reject)=>{
-            let retries = 20;
+    public async waitTxMined(hash: string): Promise<void> {
+        return new Promise<void>((resolve, reject)=>{
```

Comment:
There is something terribly wrong that we don't get proper callbacks after the call. I'll make an issue, this should work. I guess polling will work for now

Reply:
I don't like the internal polling either, but the built-in approach in metamask is also mediocre. Eg., there's a race condition when the transaction gets sent before we subscribe for the newly sent txhash.

I don't think that's what's affecting us, but if anything I'd like to expand our method to use event subscriptions with polling as fallback, instead of switching back to the web3 code - I just don't trust it anymore.

Or maybe upgrade to web3 1.0. I don't know.